src/app/filter/filter.service.spec.ts

```diff
@@ -13,4 +13,37 @@ describe('FilterService', () => {
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  it('should have values$ Observable', () => {
+    expect(service.values$).toBeDefined();
+  });
+
+  it('should have the right values', () => {
+    service.values$.subscribe(values => {
+      expect(values).toEqual({
+        currentPage: 1,
+        entriesPerPage: 20
+      });
+    });
+  });
+
+  it('should have setState method', () => {
+    expect(service.setState).toBeDefined();
+  });
+
+  it('should update state successfully', () => {
+    service.setState({
+      currentPage: 2,
+      entriesPerPage: 50,
+      nameFilter: 'test'
+    });
```

Comment:
before setting a new state you could check/expect the current status of this "state", just to be sure that you are changing it.